src/managed/OpenLiveWriter.BlogClient/Clients/WordPressClient.cs

```diff
@@ -53,5 +54,49 @@ protected override void ConfigureClientOptions(BlogClientOptions clientOptions)
             clientOptions.SupportsAuthor = true;
         }
 
+        public override BlogInfo[] GetUsersBlogs()
+        {
+            TransientCredentials tc = Login();
+            return GetWPUsersBlogs(tc.Username, tc.Password);
+        }
+        private BlogInfo[] GetWPUsersBlogs(string username, string password)
+        {
+            // call method
+            XmlNode result = CallMethod("wp.getUsersBlogs",
+                new XmlRpcString(username),
+                new XmlRpcString(password, true));
+
+            try
+            {
+                // parse results
+                ArrayList blogs = new ArrayList();
+                XmlNodeList dataValues = result.SelectNodes("array/data");
+                foreach (XmlNode dataValue in dataValues)
+                {
+                    XmlNodeList blogNodes = dataValue.SelectNodes("value/struct");
+                    foreach (XmlNode blogNode in blogNodes)
+                    {
+                        // get node values
+                        XmlNode idNode = blogNode.SelectSingleNode("member[name='blogid']/value");
+                        XmlNode nameNode = blogNode.SelectSingleNode("member[name='blogName']/value");
+                        XmlNode urlNode = blogNode.SelectSingleNode("member[name='url']/value");
```

Comment:
I'll look deeper into it, but when I added my first blog, it added the "www" blog. Then I had to add another blog using the subdomain listed. Test posts worked, adding worked, and so on. I'll dive into the code here in about 10 minutes.